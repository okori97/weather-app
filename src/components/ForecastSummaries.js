import React from "react";
import ForecastSummary from "./ForecastSummary";
import "../styles/ForecastSummaries.css";

export default function ForecastSummaries({ forecasts }) {
  return (
    <div className="forecastSummaries--list">
      {forecasts.map((forecast) => (
        <ForecastSummary
          key={forecast.date}
          date={forecast.date}
          description={forecast.description}
          icon={forecast.icon}
          temperature={forecast.temperature}
        />
      ))}
    </div>
  );
}
