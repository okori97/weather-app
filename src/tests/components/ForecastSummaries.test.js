import ForecastSummaries from "../../components/ForecastSummaries";
import React from "react";
import { render } from "@testing-library/react";

describe("forecastSummaries", () => {
  const validProp = {
    forecasts: [
      {
        date: 111,
        description: "stubDescription",
        icon: "800",
        temperature: {
          max: 10,
          min: 10,
        },
      },
      {
        date: 222,
        description: "stubDescription",
        icon: "800",
        temperature: {
          max: 10,
          min: 10,
        },
      },
    ],
    onForecastSelect: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProp.forecasts}
        onForecastSelect={validProp.onForecastSelect}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of list items", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries
        forecasts={validProp.forecasts}
        onForecastSelect={validProp.onForecastSelect}
      />
    );

    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
