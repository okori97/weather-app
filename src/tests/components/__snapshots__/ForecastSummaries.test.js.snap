// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forecastSummaries it renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecastSummaries--list"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu 1st Jan
      </div>
      <img
        alt="weather icon"
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      />
      <div
        class="forecast-summary__temperature"
      >
        10°c
      </div>
      <div
        class="forecast-summary__description"
      >
        stubDescription
      </div>
      <div
        class="forecast-summary__link-wrap"
      >
        <button
          class="forecast-summary__link"
          href="#"
          target="blank"
        >
          More details
        </button>
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu 1st Jan
      </div>
      <img
        alt="weather icon"
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      />
      <div
        class="forecast-summary__temperature"
      >
        10°c
      </div>
      <div
        class="forecast-summary__description"
      >
        stubDescription
      </div>
      <div
        class="forecast-summary__link-wrap"
      >
        <button
          class="forecast-summary__link"
          href="#"
          target="blank"
        >
          More details
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
